hack/generate-internal-groups.sh

```diff
@@ -0,0 +1,141 @@
+# user-provided API server based on k8s.io/apiserver.
+
+if [ "$#" -lt 5 ] || [ "${1}" == "--help" ]; then
+  cat <<EOF
+Usage: $(basename "$0") <generators> <output-package> <internal-apis-package> <extensiona-apis-package> <groups-versions> ...
+
+  <generators>        the generators comma separated to run (deepcopy,defaulter,conversion,client,lister,informer,openapi) or "all".
+  <output-package>    the output package name (e.g. github.com/example/project/pkg/generated).
+  <int-apis-package>  the internal types dir (e.g. github.com/example/project/pkg/apis).
+  <ext-apis-package>  the external types dir (e.g. github.com/example/project/pkg/apis or githubcom/example/apis).
+  <groups-versions>   the groups and their versions in the format "groupA:v1,v2 groupB:v1 groupC:v2", relative
+                      to <api-package>.
+  ...                 arbitrary flags passed to all generator binaries.
+
+Examples:
+  $(basename "$0") all                           github.com/example/project/pkg/client github.com/example/project/pkg/apis github.com/example/project/pkg/apis "foo:v1 bar:v1alpha1,v1beta1"
+  $(basename "$0") deepcopy,defaulter,conversion github.com/example/project/pkg/client github.com/example/project/pkg/apis github.com/example/project/apis     "foo:v1 bar:v1alpha1,v1beta1"
+EOF
+  exit 0
+fi
+
+GENS="$1"
+OUTPUT_PKG="$2"
+INT_APIS_PKG="$3"
+EXT_APIS_PKG="$4"
+GROUPS_WITH_VERSIONS="$5"
+shift 5
+
+export GO111MODULE=on
+
+go install k8s.io/code-generator/cmd/defaulter-gen \
+    k8s.io/code-generator/cmd/client-gen \
+    k8s.io/code-generator/cmd/lister-gen \
+    k8s.io/code-generator/cmd/informer-gen \
+    k8s.io/code-generator/cmd/deepcopy-gen \
+    k8s.io/code-generator/cmd/openapi-gen \
+    k8s.io/code-generator/cmd/conversion-gen \
+    k8s.io/code-generator/cmd/defaulter-gen \
+    github.com/kubernetes-incubator/apiserver-builder-alpha/cmd/apiregister-gen
+
+function codegen::join() { local IFS="$1"; shift; echo "$*"; }
+
+APIS_DIR=("${INT_APIS_PKG}/...")
+CLIENT_OUTPUT_PKG=("${OUTPUT_PKG}/client")
+
+# enumerate group versions
+ALL_FQ_APIS=() # e.g. k8s.io/kubernetes/pkg/apis/apps k8s.io/api/apps/v1
+INT_FQ_APIS=() # e.g. k8s.io/kubernetes/pkg/apis/apps
+EXT_FQ_APIS=() # e.g. k8s.io/api/apps/v1
+for GVs in ${GROUPS_WITH_VERSIONS}; do
+  IFS=: read -r G Vs <<<"${GVs}"
+  IFS=. read -r GROUP_PATH SUFFIX <<<"${G}"
+
+  if [ -n "${INT_APIS_PKG}" ]; then
+    ALL_FQ_APIS+=("${INT_APIS_PKG}/${GROUP_PATH}")
+    INT_FQ_APIS+=("${INT_APIS_PKG}/${GROUP_PATH}")
+  fi
+
+  # enumerate versions
+  for V in ${Vs//,/ }; do
+    ALL_FQ_APIS+=("${EXT_APIS_PKG}/${GROUP_PATH}/${V}")
+    EXT_FQ_APIS+=("${EXT_APIS_PKG}/${GROUP_PATH}/${V}")
+  done
+done
+
+# apiregister-gen generate the REST boilerplate of resources and their internal version
+if [ "${GENS}" = "all" ] || grep -qw "registry" <<<"${GENS}"; then
+  echo "Generating api registries"
+  "${GOPATH}/bin/apiregister-gen" --input-dirs "$(codegen::join , "${APIS_DIR[@]}")" "$@"
+fi
```

Comment:
except this, do we have other changes?
when we can vendor code-generator root directory, I'd like to use scripts from code-generator.

Reply:
I change the client output package in case of collision with the generated code of CRD and change the package path from `a.b.c` to `a` so that the behavior is consistent with higher version of kubernetes.

I'd like to change this script to call the vendored script in the code-generator directory after we upgrade the k8s dependencies to `v1.16`